pkg/apis/servicecatalog/types.go

```diff
@@ -266,10 +266,16 @@ type InstanceSpec struct {
 	// provisioned from.
 	PlanName string
 
-	// Parameters is a YAML representation of the properties to be
+	// Parameters is set of properties to be
 	// passed to the underlying broker.
 	Parameters *runtime.RawExtension
 
+	// Optional: AlphaSecretParameters is reference to the secret object containing
+	// sensitive information to pass to the underlying broker. This may be
+	// empty if no secret object is specified.
+	// +optional
+	AlphaSecretParameters *v1.LocalObjectReference
```

Comment:
In our use case the Secret the reference to which is passed in Instance, will be a Secret specifically made for Instance.

Whether Instance will handle Secret changes with propagating it to broker with [update request](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md#updating-a-service-instance) is an open question we need to discuss at some point. That said, I don't think it impacts the choice of Secret as storage for sensitive data. As an example, Pods don't get restarted when Secret with environment variables is updated, and still Pods don't embed secrets because of that.

> require all of the data to be included in the creation of the Instance/Binding and then we create the secret (for the private bits) automatically for them

I think that this approach is too specific and limiting, and hard to implement properly. It also doesn't feel like "kube way" to me - in Kubernetes world it's preferable to have independent blocks for flexibility.
Your proposal is similar to the problem we had with embedding PodPresetTemplate into Binding finally leading to the separate resource proposal https://github.com/kubernetes-incubator/service-catalog/issues/994.

Using Secret provided by user is an elegant, simple and generic solution. It was also very straightforward to implement in code. I really think it's the best solution we came up with so far, generic enough to serve any possible use case.

P.S. Shall we move the discussion back to the https://github.com/kubernetes-incubator/service-catalog/issues/621? It's hard to keep track of all the design discussions in Service Catalog.